lib/spirv/lib/src/transpiler.dart

```diff
@@ -0,0 +1,940 @@
+    final String vector1Name = resolveName(readWord());
+    // ignore second vector
+    position++;
+
+    out.write('$indent$type $name = $type(');
+
+    final int count = nextPosition - position;
+    for (int i = 0; i < count; i++) {
+      final int index = readWord();
+      out.write('$vector1Name[$index]');
+      if (i < count - 1) {
+        out.write(', ');
+      }
+    }
+    out.writeln(');');
+  }
+
+  void opCompositeConstruct() {
+    final String type = resolveType(readWord());
+    final String name = resolveName(readWord());
+    out.write('$indent$type $name = $type(');
+    final int count = nextPosition - position;
+    for (int i = 0; i < count; i++) {
+      out.write(resolveName(readWord()));
+      if (i < count - 1) {
+        out.write(', ');
+      }
+    }
+    out.writeln(');');
+  }
+
+  void opCompositeExtract() {
+    final String type = resolveType(readWord());
+    final String name = resolveName(readWord());
+    final String src = resolveName(readWord());
+    out.write('$indent$type $name = $src');
+    final int count = nextPosition - position;
+    for (int i = 0; i < count; i++) {
+      final int index = readWord();
+      out.write('[$index]');
+    }
+    out.writeln(';');
+  }
+
+  void opFNegate() {
+    final String type = resolveType(readWord());
+    final String name = resolveName(readWord());
+    final String operand = resolveName(readWord());
+    out.writeln('$indent$type $name = -$operand;');
+  }
+
+  void opLabel() {
+    out.writeln('{');
+    indent = indent + '  ';
+    if (target == TargetLanguage.sksl && currentFunction == entryPoint) {
+      final String ind = indent;
+      if (fragCoord > 0) {
+        final String fragName = resolveName(fragCoord);
+        out
+          ..write(ind)
+          ..writeln('float4 $fragName = float4($_fragParamName, 0, 0);');
+      }
+      out
+        ..write(ind)
+        ..writeln('float4 $_colorVariableName;');
+    }
+  }
+
+  void opReturn() {
+    if (currentFunction == entryPoint) {
+      return;
+    }
+    out.writeln(indent + 'return;');
+  }
+
+  void opReturnValue() {
+    final String name = resolveName(readWord());
+    out.writeln(indent + 'return $name;');
+  }
+
+  void parseOperatorInst(String op) {
+    final String type = resolveType(readWord());
+    final String name = resolveName(readWord());
+    final String a = resolveName(readWord());
+    final String b = resolveName(readWord());
+    out.writeln('$indent$type $name = $a $op $b;');
+  }
+
+  void parseBuiltinFunction(String functionName) {
+    final String type = resolveType(readWord());
+    final String name = resolveName(readWord());
+    out.write('$indent$type $name = $functionName(');
+    final int count = nextPosition - position;
+    for (int i = 0; i < count; i++) {
+      out.write(resolveName(readWord()));
+      if (i < count - 1) {
+        out.write(', ');
+      }
+    }
+    out.writeln(');');
+  }
+
+  void parseGLSLInst(int id, int type) {
```

Comment:
For a switch statement of this size, a lookup in a constant map may be faster and/or more compact. Like:
```dart
const Map<int, String> _glslStd450OpNames = const <int, String>{
  _glslStd450Trunc: 'trunc',
  ...
};

const Map<int, String> _glslStd450OpArgc = const <int, int>{
  _glslStd450Trunc: 1,
  ...
};

...

void parseGLSLInst(int id, int type) {
  final int inst = readWord();
  final String opName = _glslStd450OpNames[inst];
  // throw if opName is null
  final int opArgc = _glslStd450OpArgc[inst];
  parseGLSLOp(id, type, opName, opArgc);
}
```